use-cases/user-expectations/6/expectations-6.md

```diff
@@ -0,0 +1,76 @@
+### Use Case #6
+
+Name: Synchronous exception in Promise constructor always rejects even if promise is already resolved.
+
+Brandy is writing code with Promises. She writes the following program, and is surprised/confused by not seeing the `"resolved!"` message in the console. That confusion causes her to worry that the Promise constructor is not even running, even though it seems like `showSuccess()` is being called.
+
+```js
+var OK = true;
+
+var p = new Promise(function c(resolve,reject){
+   if (OK) {
+      resolve("Good to go!");
+      cosnole.log("resolved!");   // console misspelled here
+   }
+   else {   
+      reject("Oops!");
+      console.log("rejected.");
+   }
+});
+
+p.then( showSuccess, showError );
+// ..nothing printed to console :(
+```
+
+### What happens
+
+The promise is already fulfilled (with `resolve(..)`), so the exception that then happens from `cosnole.log(..)` being misspelled is silently swallowed.
+
+### Why it happens
+
+Promises can only be resolved to a fulfillment or rejection, and once resolved, cannot be changed. This normally makes sense, but in this specific case of an accidental exception that occurs synchronously subsequent to the otherwise resolution (fulfillment or rejection), the user never expects the promise machinery to just suppress/hide that exception.
+
+### What can we maybe do better
```

Comment:
I may not fully know what an "async_hook" is, but I shyed away from that because this use-case is specifically about synchronous behavior (both the resolution and the exception overriding it).

Having a hook to be notified of these cases is not a bad idea, but I organized this use-case under "expectations" because I think most users would tend to just expect this suggested behavior. So a hook would sorta be augmentative of that rather than an alternative.